app/api_contract.py

```diff
@@ -0,0 +1,385 @@
+    ),
+]
+
+CloudUserToken = Annotated[
+    str,
+    Header(
+        alias="Authroization",
+        title="FinOps for Cloud user token",
+        description="The token used for authorization in `Bearer <token>` format",
+    ),
+]
+
+
+class ErrorResponse(BaseModel):
+    detail: str
+
+
+class CommonPrivateFields:
+    id: uuid.UUID
+    created_at: datetime.datetime
+    updated_at: datetime.datetime
+    deleted_at: Annotated[
+        datetime.datetime | None,
+        Field(
+            default=None,
+            examples=[None],
+            description="If not null, the timestamp the object was deleted at",
+        ),
+    ]
+
+
+class CreateOrgRequest(BaseModel):
+    display_name: Annotated[str, Field(examples=["Apple Inc."])]
+
+
+class OrganizationDetail(BaseModel, CommonPrivateFields):
+    display_name: Annotated[str, Field(examples=["Apple Inc."])]
+    external_id: Annotated[
+        str,
+        Field(
+            description="the ID of the account of the MPT platform which owns this organization"
+        ),
+    ]
+    cloudspend_id: uuid.UUID
+
+
+class CreateUserRequest(BaseModel):
+    email: EmailStr
+    organization_id: uuid.UUID
+    display_name: Annotated[str, Field(examples=["Tim Cook"])]
+
+
+class UserDetail(BaseModel, CommonPrivateFields):
+    email: EmailStr
+    organization_id: uuid.UUID
+    display_name: Annotated[str, Field(examples=["Tim Cook"])]
+    cloudspend_id: uuid.UUID
+
+
+class CloudType(enum.Enum):
+    AWS_ROOT = "aws_root"
+    AWS_LINKED = "aws_linked"
+    AZURE_TENANT = "azure_tenant"
+    AZURE_SUBSCRIPTION = "azure_subscription"
+    GCP = "gcp"
+    ALIBABA = "alibaba"
+    DATABRICKS = "databricks"
+    KUBERNETES = "kubernetes"
+
+
+class AWSExportType(enum.Enum):
+    STANDARD_DATA_EXPORT = "standard_data_export"
+    LEGACY_COST_AND_USAGE = "legacy_cost_and_usage"
+
+
+class AWSExportConfig(BaseModel):
+    create_new_data_export: Annotated[
+        bool,
+        Field(
+            default=True,
+            description=(
+                "If set to True, a new data export will be created, "
+                "otherwise only connect to data in bucket"
+            ),
+        ),
+    ]
+    export_name: str
+    export_s3_bucket_name: str
+    export_path_prefix: str
+
+
+class AWSRootAccountConfig(BaseModel):
+    aws_access_key_id: str
+    aws_secret_access_key: str
+    use_aws_edp: Annotated[
+        bool, Field(default=False, description="Use AWS Enterprise Discount Program?")
+    ]
+    export_type: Annotated[
+        AWSExportType, Field(default=AWSExportType.STANDARD_DATA_EXPORT)
+    ]
+    export_config: Annotated[
+        AWSExportConfig | None,
+        Field(
+            default=None,
+            description=(
+                "If set to None, the existing data exports will be automatically detected, "
+                "otherwise use these fields to manually configure the export"
+            ),
+        ),
+    ]
```

Comment:
This is quite complicated, copied it from the UI, I assume we probably don't need most of these but included everything for completeness and as a starting point for a discussion